src/lib/registration/sw_mgmt.rb

```diff
@@ -121,12 +121,16 @@ def self.find_base_product
 
       # use the selected product if a product has been already selected
       selected = product_selected? if Stage.initial
+      selected = !AutoinstConfig.selected_product.nil? if Mode.auto
 
       # during installation the products are :selected,
       # on a running system the products are :installed
       # during upgrade use the newer selected product (same as in installation)
       products = Pkg.ResolvableProperties("", :product, "").find_all do |p|
-        if Stage.initial
+        if Stage.initial && Mode.auto
+          Yast.import "AutoinstConfig"
+          p["short_name"] == AutoinstConfig.selected_product.short_name
```

Comment:
Sorry, I have to repeat again:

Is it really `short_name`? That is a name suitable for displaying in the UI, for product identification we should use the `name` attribute. The short name usually includes a SP level (e.g. `SLES12-SP3`) that means the profile would not work with the SP+1 version and would need to be manually adjusted.

Or is there any reason for `short_name`?

Reply:
no, I've just used that during testing ... I'll clean that.

Reply:
btw ```selected_product``` is a base product available on media. In fact it is ```Y2Packager::Product``` so it doesn't depend much if the selection is done using ```name``` or ```short_name```